ipa-core/src/protocol/ipa_prf/aggregation/bucket.rs

```diff
@@ -131,9 +140,21 @@ where
             let left_index = index * step;
             let right_index = left_index + span;
 
-            row_contribution[left_index] -= &bdbit_contribution;
+            // bdbit_contribution is either zero or equal to row_contribution. So it
+            // is okay to do a carryless "subtraction" here.
+            for (r, b) in row_contribution[left_index]
+                .iter_mut()
+                .zip(bdbit_contribution.iter())
+            {
+                *r -= b;
+            }
             if right_index < breakdown_count {
-                row_contribution[right_index] = bdbit_contribution;
+                for (r, b) in row_contribution[right_index]
+                    .iter_mut()
+                    .zip(bdbit_contribution)
+                {
+                    *r = b;
+                }
```

Comment:
This part also seems like it could be simplified if we stopped using BitDecomposed and used BooleanArray instead.